pkg/tests/tasks/traffic/ingress/gatewayapi_test.go

```diff
@@ -0,0 +1,164 @@
+package ingress
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/maistra/maistra-test-tool/pkg/app"
+	"github.com/maistra/maistra-test-tool/pkg/tests/ossm"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/assert"
+	"github.com/maistra/maistra-test-tool/pkg/util/env"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/pod"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
+	"github.com/maistra/maistra-test-tool/pkg/util/shell"
+	. "github.com/maistra/maistra-test-tool/pkg/util/test"
+	"github.com/maistra/maistra-test-tool/pkg/util/version"
+)
+
+func TestGatewayApiControllerMode(t *testing.T) {
+	NewTest(t).Id("T41").Groups(Full, InterOp).Run(func(t TestHelper) {
+		if env.GetSMCPVersion().LessThan(version.SMCP_2_3) {
+			t.Skip("TestGatewayApiControllerMode was added in v2.3")
+		}
+		ns := "foo"
+		t.Cleanup(func() {
+			oc.RecreateNamespace(t, ns)
+		})
+
+		ossm.DeployControlPlane(t)
+
+		smcpName := "basic"
+
+		t.LogStep("Install Gateway API CRD's")
+		shell.Executef(t, "kubectl get crd gateways.gateway.networking.k8s.io &> /dev/null ||   { kubectl kustomize github.com/kubernetes-sigs/gateway-api/config/crd/experimental?ref=v0.5.1 | kubectl apply -f -; }")
+
+		oc.CreateNamespace(t, ns)
+
+		t.LogStep("Install httpbin")
+		app.InstallAndWaitReady(t, app.Httpbin(ns))
+		t.Cleanup(func() {
+			app.Uninstall(t, app.Httpbin(ns))
+		})
+
+		t.LogStep("Deploy the Gateway app")
+		oc.ApplyString(t, ns, gatewayDeployment)
+		t.Cleanup(func() {
+			oc.DeleteFromString(t, ns, gatewayDeployment)
+		})
+
+		t.LogStep("Deploy the Gateway SMCP")
+		if env.GetSMCPVersion().LessThan(version.SMCP_2_4) {
+			oc.Patch(t, meshNamespace, "smcp", smcpName, "merge", `
+        spec:
+          runtime:
+            components:
+              pilot:
+                container:
+                  env:
+                    PILOT_ENABLE_GATEWAY_API: “true”
+                    PILOT_ENABLE_GATEWAY_API_STATUS: “true”
+                    PILOT_ENABLE_GATEWAY_API_DEPLOYMENT_CONTROLLER: “true”`)
+			t.Cleanup(func() {
```

Comment:
You can put this also inside the first cleanup and maybe instead of set the entire patch you can set some like this:
```
cleanGatewayString := ""
```

And then inside each loop set `cleanGatewayString` and inside the cleanup set
```
oc.Patch(t, meshNamespace, "smcp", smcpName, "json", cleanGatewayString)
```

Reply:
WDYT?

Reply:
Actually, both paths are different. 

If you observe 2.3 celanup ends with runtime
2.4 cleanup ends with techPreview

this won't work with the single string